pkg/osdctlConfig/osdctlConfig.go

```diff
@@ -0,0 +1,85 @@
+package osdctlConfig
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/adrg/xdg"
+)
+
+const (
+	ConfigFileName = "osdctl"
+)
+
+// Generates the config file path for osdctl config
+func GenerateConfigFilePath() (string, error) {
+	configFilePath, err := xdg.ConfigFile(ConfigFileName)
+	if err != nil {
+		return "", err
+	}
+	return configFilePath, nil
+}
+
+func EnsureConfigFile() error {
+	configFilePath, err := GenerateConfigFilePath()
+	if err != nil {
+		return err
+	}
+
+	if _, err := os.Stat(configFilePath); errors.Is(err, os.ErrNotExist) {
+		_, err = os.Create(configFilePath)
+		if err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+func GetValueForKey(key string) (string, error) {
+
+	configFileMap, err := generateMapFromConfig()
+	if err != nil {
+		return "", err
+	}
+
+	if value, ok := configFileMap[key]; ok {
+		return strings.TrimSpace(value), nil
```

Comment:
We should probably move the trimming of key and value into the generateMapFromConfig after the split, to avoid people accidentally setting an invalid kvs format and messing up

Reply:
I agree, it's probably best to contain all parsing logic in one place. Changing